cpp/dolfin/fem/Form.cpp

```diff
@@ -74,6 +81,40 @@ int Form::original_coefficient_position(int i) const
   return _coefficients.original_position(i);
 }
 //-----------------------------------------------------------------------------
+void Form::set_constants(
```

Comment:
Is there a case where we could pass `Constant`s to the constructor,  and then call `set_constants`? If no, the check in this function would be much simpler.

Reply:
Indeed. In command-line workflow, a Form is constructed with vector of constants, but only their names are known at that point, pointers to the actual constants are `nullptr`. Later on, in user's code, `form.set_constants` is called to match the names to actual `Constant`s that user prepared.
This is exactly the same as what happens for `Coefficient`.

Reply:
Perhaps it would be better to set all coefficients and constants post-construction for consistency.